File: js-app/admin-frontend/app/models/activity.js
import DS from 'ember-data';

export default DS.Model.extend({
  user: DS.belongsTo('user', {async: true}),
  device: DS.belongsTo('device', {async: true}),

  statusTo: DS.attr('string'),
  statusFrom: DS.attr('string'),
  createdAt: DS.attr('date')
});

File: app/models/activity.rb
class Activity < ActiveRecord::Base
  VALID_STATUSES = %w(available in_maintenance checked_out)

  belongs_to :user
  belongs_to :device

  validates :user,
            presence: true

  validates :device,
            presence: true

  validates :status_from,
            presence: true,
            inclusion: {in: VALID_STATUSES}

  validates :status_to,
            presence: true,
            inclusion: {in: VALID_STATUSES}
end
